perun-core/src/main/resources/oracleChangelog.txt

```diff
@@ -6,6 +6,60 @@
 -- Directly under version number should be version commands. They will be executed in the order they are written here.
 -- Comments are prefixed with -- and can be written only between version blocks, that means not in the lines with commands. They have to be at the start of the line.
 
+3.1.47
+delete from service_dependencies where dependency_id in (select id from exec_services where exec_services.type='GENERATE');
+delete from service_denials where exec_service_id in (select id from exec_services where exec_services.type='GENERATE');
+delete from tasks where exec_service_id in (select id from exec_services where exec_services.type='GENERATE');
+delete from exec_services where type='GENERATE';
+alter table tasks drop constraint TASK_STAT_CHK;
+update tasks set status='ERROR' where status != 'ERROR' and status != 'DONE';
+alter table tasks add constraint TASK_STAT_CHK check (status in ('WAITING', 'PLANNED', 'GENERATING', 'GENERROR', 'GENERATED', 'SENDING', 'DONE', 'SENDERROR', 'ERROR'));
+alter table services add column description nvarchar2(1024);
+alter table services add column delay integer default 10 not null;
+alter table services add column recurrence integer default 2 not null;
+alter table services add column enabled char(1) default '1' not null;
+alter table services add column script nvarchar2(256);
+update services set script=exec_services.script from exec_services where services.id=exec_services.service_id;
+update services set enabled=exec_services.enabled from exec_services where services.id=exec_services.service_id;
+alter table services alter column script set not null;
+update services set recurrence=2 , delay=10;
+alter table service_denials add column service_id integer;
+update service_denials set service_id=temp.id from (select id as exec_service_id, service_id as id from exec_services) temp where temp.exec_service_id=service_denials.exec_service_id;
+alter table service_denials add constraint SRVDEN_SRV_FK foreign key (service_id) references services(id);
+create index IDX_FK_SRVDEN_SRV on service_denials(service_id);
+alter table service_denials drop constraint SRVDEN_U;
+drop index IDX_SRVDEN_U;
```

Comment:
Bugged that doesn't exists, probably dropped by DB when we drop constraint above.